### DISCUSSION TITLE
Variables in build scripts

### DISCUSSION THREAD

**Discussion by mmh352** [mmh352/ou-container-builder#20] [General]
Just in passing, I also note that in my original Dockerfiles, the version (3.3) was assigned to a `VERSION` variable and referenced as such. Again, the current config does not support that, but I know from experience that if you leave version numbers in paths as literals, when you update a version it's all too easy to miss one!

_Originally posted by @psychemedia in https://github.com/mmh352/ou-container-builder/issues/11#issuecomment-832075036_

**Comment by mmh352** [comment]
Variables within an inline script will work. Variables within the ContainerConfig file currently don't work. It is an interesting suggestion. I think I would need to see a concrete example where that is needed.

The ideal safety net is of course to have some automated testing of the container.

**Comment by psychemedia** [comment]
Re: variables, another useful variable may something along the lines of the Jupyter official stack convention of `$NB_USER`, as discussed more in https://github.com/mmh352/ou-container-builder/issues/19.

**Comment by mmh352** [comment]
Moved this into a discussion, because it is really more of a long-term discussion, rather than a specific, concrete issue that needs fixing.

**Comment by psychemedia** [comment]
Re: `Variables within an inline script will work`

It would be good to maintain a list somewhere (docs?) of the variables that are reusable within the inline scripts.

For example, assuming that ENV variables are accessible, I think this includes (check if it does):

- [ ] `USER` (default: `ou-user`)
- [ ]  `UID` (default: `1000`)
- [ ]  `GID` (default: `100`)
- [ ]  `MODULE_CODE` (default: `{{ module.code }}`)
- [ ]  `MODULE_PRESENTATION` (default: `{{ module.presentation }}`)
- [ ]  `HOME` (default: `/home/$USER/$MODULE_CODE-$MODULE_PRESENTATION`)

**Reply by mmh352** [reply]
That is a good idea. I have factored it out into its own issue. The documentation in general needs to be improved.

**Reply by psychemedia** [reply]
The checkboxes were dropped in the move to the issue. Also, should they be checked (i.e. are those environment variables available to inline scripts?)

**Reply by mmh352** [reply]
I have no idea :-). I need to check that first, before I update the documentation.
